lib/util/git/index.spec.ts

```diff
@@ -56,9 +57,9 @@ describe('util/git/index', { timeout: 10000 }, () => {
     await fs.writeFile(base.path + '/master_file', defaultBranch);
     await fs.writeFile(base.path + '/file_to_delete', 'bye');
     await repo.add(['master_file', 'file_to_delete']);
-    await repo.commit('master message', [
-      '--date=' + masterCommitDate.toISOString(),
-    ]);
+    process.env.GIT_COMMITTER_DATE = masterCommitDate.toISOString();
```

Comment:
why this change?

Reply:
I changed the implementation to rely on git committer date rather than rely on the author date.

The committer date is a lot more reliable for this use case given it reflects the date that commit was lastly changed (amended for example), as opposed to reflecting the date where the commit was originally authored.

For example, a commit could be authored 1 year ago and still being amended, cherry-picked, or rebased, and the committer date is what will reflect that.

In tests, to override the committer date, an environment variable `GIT_COMMITTER_DATE` is required. There is no flag for it because it is not meant to be easily overridden.

Reply:
I can extract that to a separate PR if you prefer.